### DISCUSSION TITLE
Turn off/on ERG mode

### DISCUSSION THREAD

**Discussion by Lory3881** [dvmarinoff/Auuki#222] [General]
Is it possible to switch between ERG mode and SIM mode while doing a structured workout? To clarify better: I would like to do a structured workout choosing to do it in ERG mode or in SIM mode.

**Comment by dvmarinoff** [comment]
Currently you can do:

1. Slope targets

define a workout with Slope targets which control the resistance and secondary ERG power that just shows in the interface as a target but does not control the resistance.
You can mix and match on interval bases, on interval can have a Slope target that controls the resistance the next one could be ERG.
You can also switch the current interval to another mode, but you can't switch the whole workout from ERG to Slope mode, it has to be predefined and potentially manually overwritten during the ride if needed, unless ...

2. Dual workouts
Since you can switch most things on the fly, like FTP as a way to control intensity, you can also switch the workouts on the fly.
Therefore if you have two identical workouts one with ERG target and one with Slope targets you can switch between them.
The change will take effect from the start of the next interval.

I am not sure which case is yours.

Here is how to define slope targets: https://github.com/dvmarinoff/Auuki/wiki/Slope-target-for-workout-intervals

**Comment by kuuhp** [comment]
Hi, I want to second this idea (if I got it right):

I really like the ERG mode for longer intervals (>2min). However, when I'm doing shorter intervals (e.g. 30/30s) I'd like to switch of ERG mode and simply work against some sort of resistance/slope. This way I don't lose seconds when the trainer is adjusting (plus I can go beyond the target for all-out-efforts). After the intervals I usually switch back to ERG mode to do the recovery block.

Currently you can turn off ERG mode in a structured workout, but it gets automatically turned back on with the next workout block.

Is there any chance this can make it into the backlog? :-)

**Reply by dvmarinoff** [reply]
I guess I can add a lock button for the current mode, so that if you press lock the mode you select will take precedence on the mode programmed in the workout. Will that work?

**Reply by kuuhp** [reply]
I guess that would work. It could basically be a switch that toggles between automatic mode (i.e. setting the workout targets for ERG, slope, resistance) and manual mode (the manual setting on the slope/resistance tab overwrites the targetsetting from the workout). It is important however, that the workout with its steps is still shown correctly to be able to follow along.

I hope that maked sense...

**Comment by dvmarinoff** [comment]
Just added a Lock Button that can lock the current trainer control mode and prevent automated changes:
<img width="1200" height="675" alt="mode-lock-toggle-x-post" src="https://github.com/user-attachments/assets/bc13082e-5936-4cf0-9b9c-3dc05002360e" />

**Reply by kuuhp** [reply]
I was on vacation and the weather was simply too good to drive indoors. I tested the feature last week and it works as intended, thank you.

One remark: Currently the lock works on trainer control mode level. You might want to think about to lock the current settings within the trainer control mode. I.e. I want to look the current ERG settings instead of having the software switch to the next interval (because of exhaustion, ad-hoc changes in training, ...).

**Reply by dvmarinoff** [reply]
Which would fit better:
1. Extend the current interval duration, and move to the next only after the Lap button is tapped
2. Lock the current magnitude value for the currently selected mode into the next interval
3. Pause the workout but keep recording until the workout is unpaused

Option 1 may make more sense, including for extending warmups and cooldowns. 
Option 2 would be the easiest to add. 
Option 3 might result in a complex confusion UI with separate controls for running a workout and recording an activity

**Reply by kuuhp** [reply]
While everybody will have different use cases I'd prefer option 2: I'd use the feature to skip a certain interval to get more recovery-time.

Thinking about it: The lock-icon would then basically turn off the automatic control from the loaded workout and let the user manually adjust ERG (WATTs), resistance or slope. For the time the lock is on users can adjust the settings to their liking without the workout overwriting the settings. All while the workout keeps running and is being recorded. Does that make any sense?

I have to add: My original use case does work with the current version. I just stumbled upon this as the software did not work as I expected.

Thank you again.

**Reply by dvmarinoff** [reply]
Makes sense. I am thinking of adding a second lock for the resistance magnitude. One that will sit on the +/- control row, just under the mode lock. You'd need to press both for this particular case, but would be more intuitive in my opinion, also adds more granular control options. On the down side it's going to look like the properties section of a desktop popup menu.

**Reply by kuuhp** [reply]
Thinking from a usability point of you: Instead of locking something the icon can basically be a toggle between automatic control (from the workout) and manual control. Automatic mode is fed by the loaded workout, while manual lets you chose the mode and the magnitude for ERG, resistance and slope. For this you would only need one icon (maybe a [m] / [a] in some sort of way).
